src/common/bip32.h

```diff
@@ -10,7 +10,7 @@
  * 0  - MAX_DERIVATION_INDEX
  * 0' - MAX_DERIVATION_INDEX'
  */
-#define MAX_DERIVATION_INDEX 512
+#define MAX_DERIVATION_INDEX 0x100000u
```

Comment:
Why is it in hex? I feel an integer is easier to read in this case.

Reply:
The main reason is that hardened derivations means that the index is greater than `0x10000000u` (or 268435456) and in hex it's easier to spot that this doesn't allows hardened derivation.

Reply:
It seems very confusing to be counting zeros. I think we should put 512 and explain that only non-hardened derivations will be allowed.

Reply:
Great! Thanks!